src/cmds/pbs_rstat.c

```diff
@@ -100,7 +101,7 @@ display_single_reservation(struct batch_status *resv, int how)
 			else if (strcmp(attrp->name, ATTR_resv_end) == 0)
 				resv_end = attrp->value;
 			else if (strcmp(attrp->name, ATTR_resv_duration) == 0)
-				resv_duration = atoi(attrp->value);
+				resv_duration = strtol(attrp->value, NULL, 10);
```

Comment:
Well initially Sudeshna added a check and was printing an error on stderr but the thing is that this is a command, so printing an error message in the middle of a defined output will break compatibility with tests/tool that might be dependent on its output. 
We decided not to check for the error case because the likelihood of value fed to strtol (which is coming from pbs server) being non-numeric is remote. PBS server has checks and balances in place to accept only numeric durations.

Reply:
Thanks for clarifying Arun. So, does pbs_rstat not do any client side sanity checking? It seems weird to me that we leave such easy to do checks to the server. If it really doesn't do any client side checks, then I agree with you, we should leave this alone.

Reply:
I don't think it does, @SudeshnaMoulik can comment on it since she has seen the code. I do not see it as leaving the checks to the server case because the server anyways has to do checks before accepting inputs. Making the checks in client would be very useful if it was getting information from an untrusted source of data.

Reply:
Ok, I saw the code in a bit more detail, this information should already be vetted by the server as it's coming from one of the stat calls, I take my comment back then, the check is not needed.